.drone.star

```diff
@@ -0,0 +1,173 @@
+config = {
+	'app': 'owncloud-sdk',
+
+	'branches': [
+		'master'
+	],
+
+	'yarnlint': True,
+
+	'build': True
+}
+
+def main(ctx):
+    return [ fullBuildAndPublish(), testWithinSubFolder() ]
+
+def incrementVersion():
+    return [{
+    		'name': 'increment-version',
+    		'image': 'node:11',
+    		'pull': 'always',
+    		'commands': [
+    			'yarn version --no-git-tag-version --new-version 1.0.0-${DRONE_BUILD_NUMBER}'
+    		],
+            'when': {
+                'event': [
+                    'push'
+                ]
+            }
+    }]
+
+def buildDocs()	:
+    return [{
+    		'name': 'build-docs',
+    		'image': 'node:11',
+    		'pull': 'always',
+    		'commands': [
+    			'yarn install',
+    			'yarn build:docs'
+    		],
+    }]
+
+def buildSystem():
+    return [{
+    		'name': 'build-system',
+    		'image': 'node:11',
+    		'pull': 'always',
+    		'commands': [
+    			'yarn install',
+    			'yarn lint',
+    			'yarn build:docs'
+    		],
+    }]
+
+def prepareTestConfig(subFolderPath = '/'):
+    return [{
+        'name': 'prepare-test-config',
+        'image': 'owncloud/ubuntu:16.04',
+		'commands': [
+			'apt update',
+			'apt install gettext -y',
+			'envsubst < tests/config/config.drone.json > tests/config/config.json',
+			'cat tests/config/config.json'
+		],
+        'environment' : {
+            'SUBFOLDER': subFolderPath
+        }
+    }]
+
+def test():
+    return [{
+        'name': 'test',
+        'image': 'owncloudci/chromium',
+        'pull': 'always',
+        'commands': [
+            'yarn test-drone'
+        ]
+    }]
+
+def publishDocs():
```

Comment:
this step will run twice, in fullBuildAndPublish and testWithinSubFolder, it should only run only once after both types of tests pass